jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala

```diff
@@ -229,30 +230,25 @@ class XGBoostClassificationModel private[ml](
     this
   }
 
-  // TODO: Make it public after we resolve performance issue
-  private def margin(features: Vector): Array[Float] = {
-    import DataUtils._
-    val dm = new DMatrix(scala.collection.Iterator(features.asXGB))
-    _booster.predict(data = dm, outPutMargin = true)(0)
-  }
-
-  private def probability(features: Vector): Array[Float] = {
-    import DataUtils._
-    val dm = new DMatrix(scala.collection.Iterator(features.asXGB))
-    _booster.predict(data = dm, outPutMargin = false)(0)
-  }
-
+  // The performance of single instance prediction is not ideal, use it carefully!
   override def predict(features: Vector): Double = {
-    throw new Exception("XGBoost-Spark does not support online prediction")
+    import DataUtils._
```

Comment:
can we logwarn here?

Reply:
This function would be called for each instance, very frequently. I think logging inside would flood log files.

Reply:
doing this? https://github.com/CodingCat/spark/blob/554cd391b3ddb5fb3f7c52950610e832ad40047b/core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala#L136-L147